src/main/scala/uk/gov/hmrc/ct/ct600/v3/calculations/LoansToParticipatorsCalculator.scala

```diff
@@ -34,38 +38,29 @@ trait LoansToParticipatorsCalculator extends CtTypeConverters {
 
   // CHRIS cardinality 1..1 - cannot be null
   def calculateA15(loans2p: LoansToParticipators): A15 = {
-    val sumOfLoanAmounts: Int = loans2p.loans.flatMap(l => Some(l.amount)).sum
+    val sumOfLoanAmounts: Int = (loans2p.loans.map(_.amount)).sum
     A15(Some(sumOfLoanAmounts))
   }
 
   // CHRIS cardinality 1..1 - cannot be null
-  def calculateA20(a15: A15): A20 = {
-    A20(a15.value.map(x => BigDecimal(x * 0.25)))
+  def calculateA20(a15: A15, loans2p: LoansToParticipators): A20 = {
+    val amountOfA15BeforeApril2016: Int = loans2p.loans.flatMap(_.amountBefore06042016).sum
+    val value = calculate(a15.value, amountOfA15BeforeApril2016)
```

Comment:
Can we rename this `calculate` function they are all sharing? Not sure what it does.